api/app/reviews/api.py

```diff
@@ -1,7 +1,134 @@
+    'user_category': fields.String(attribute='user_category.name')
+}
+
+review_response_fields = {
+    'review_form': fields.Nested(review_form_fields),
+    'response': fields.Nested(response_fields),
+    'user': fields.Nested(user_fields),
+    'reviews_remaining_count': fields.Integer
+}
+
+class ReviewResponseUser():
+    def __init__(self, review_form, response, reviews_remaining_count):
+        self.review_form = review_form
+        self.response = response
+        self.user = None if response is None else response.user
+        self.reviews_remaining_count = reviews_remaining_count
+
+class ReviewAPI(ReviewMixin, restful.Resource):
+
+    @auth_required
+    @marshal_with(review_response_fields)
     def get(self):
-        return 200
\ No newline at end of file
+        args = self.req_parser.parse_args()
+        event_id = args['event_id']
+        
+        review_form = db.session.query(ReviewForm)\
+                        .join(ApplicationForm, ApplicationForm.id==ReviewForm.application_form_id)\
+                        .filter_by(event_id=event_id)\
+                        .first()
+        if review_form is None:
+            return EVENT_NOT_FOUND
+
+        reviews_remaining_count = db.session.query(func.count(ResponseReviewer.id))\
+                        .filter_by(reviewer_user_id=g.current_user['id'])\
+                        .join(Response)\
+                        .filter_by(is_withdrawn=False, application_form_id=review_form.application_form_id, is_submitted=True)\
+                        .outerjoin(ReviewResponse, and_(ReviewResponse.response_id==ResponseReviewer.response_id, ReviewResponse.reviewer_user_id==g.current_user['id']))\
+                        .filter_by(id=None)\
+                        .all()[0][0]
+
+        skip = self.sanitise_skip(args['skip'], reviews_remaining_count)
+
+        response = db.session.query(Response)\
+                        .filter_by(is_withdrawn=False, application_form_id=review_form.application_form_id, is_submitted=True)\
+                        .join(ResponseReviewer)\
+                        .filter_by(reviewer_user_id=g.current_user['id'])\
+                        .outerjoin(ReviewResponse, and_(ReviewResponse.response_id==ResponseReviewer.response_id, ReviewResponse.reviewer_user_id==g.current_user['id']))\
+                        .filter_by(id=None)\
+                        .order_by(ResponseReviewer.response_id)\
+                        .offset(skip)\
+                        .limit(1)\
+                        .first()
+        
+        return ReviewResponseUser(review_form, response, reviews_remaining_count)
+
+    def sanitise_skip(self, skip, reviews_remaining_count):
+        if skip is None:
+            skip = 0
+
+        if skip < 0:
+            skip = 0
+
+        if reviews_remaining_count == 0:
+            skip = 0
+        elif skip >= reviews_remaining_count:
+            skip = reviews_remaining_count - 1

```

Comment:
Why set `skip = reviews_remaining_count - 1` and not `skip = reviews_remaining_count` (if Skip > reviews_remaining_count)?

Reply:
Say `reviews_remaining_count=2`, if you skip 1, you go on to the second review. If you skip 2, you get nothing back. So the subtraction is to get you to the last review.